warehouse/validations/steps.go

```diff
@@ -3,61 +3,72 @@ package validations
 import (
 	"encoding/json"
 
-	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+	schemarepository "github.com/rudderlabs/rudder-server/warehouse/integrations/datalake/schema-repository"
 	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
 )
 
-func (ct *CTHandleT) validationStepsFunc(req json.RawMessage, _ string) (json.RawMessage, error) {
-	ct.infoRequest = &DestinationValidationRequest{}
-	if err := parseOptions(req, ct.infoRequest); err != nil {
-		return nil, err
-	}
-
-	return json.Marshal(validationStepsResponse{
-		Steps: ct.validationSteps(),
-	})
+func validateStepFunc(destination *backendconfig.DestinationT, _ string) (json.RawMessage, error) {
+	return json.Marshal(StepsToValidate(destination))
 }
 
-// validationSteps returns series of validation steps for
-// a particular destination.
-func (ct *CTHandleT) validationSteps() []*validationStep {
-	steps := []*validationStep{{
-		ID:        1,
-		Name:      verifyingObjectStorage,
-		Validator: ct.verifyingObjectStorage,
+func StepsToValidate(dest *backendconfig.DestinationT) *model.StepsResponse {
+	var (
+		destType = dest.DestinationDefinition.Name
+		steps    []*model.Step
+	)
+
+	steps = []*model.Step{{
+		ID:   len(steps) + 1,
+		Name: model.VerifyingObjectStorage,
 	}}
 
-	// Time window destination contains only object storage verification
-	if misc.Contains(warehouseutils.TimeWindowDestinations, ct.infoRequest.Destination.DestinationDefinition.Name) {
-		return steps
-	}
+	switch destType {
+	case warehouseutils.GCS_DATALAKE, warehouseutils.AZURE_DATALAKE:
+		break
+	case warehouseutils.S3_DATALAKE:
+		wh := createDummyWarehouse(dest)
+		if canUseGlue := schemarepository.UseGlue(&wh); !canUseGlue {
+			break
+		}
 
-	steps = append(steps,
-		&validationStep{
-			ID:        2,
-			Name:      verifyingConnections,
-			Validator: ct.verifyingConnections,
```

Comment:
There was too much coupling with steps and validation functions. steps API will return only the steps. Once we get the steps, we call `NewValidator` to get the validating function.